### PR TITLE
both proxy and admin log do not need pass in correlation id explicitly

### PR DESCRIPTION
**Pull request by GingerMoon**
Hi @spikelu2016 , this PR changes all the correlation id log related code, both admin and proxy.
Now the sub logger with correlation id is stored inside the http request context. 
Whenever we can get it from the http request context whenever we need it, especially when we are in some deeper place. 
In such cases, we need to pass the http request context as the parameter to this "deeper place". 

Please feel free to let me know if you have any comments.

### CODE REVIEW

File: internal/server/web/proxy/run.go [hunk 1]
```diff
@@ -75,7 +73,6 @@ func logRunResponse(log *zap.Logger, data []byte, prod, private bool, cid string
 func logRetrieveRunRequest(log *zap.Logger, prod bool, cid, tid, rid string) {
```
Comment (spikelu2016) [inline internal/server/web/proxy/run.go:73]: `cid` can also be removed from the function header for all the functions inside this file

File: internal/server/web/proxy/vllm.go [hunk 2]
```diff
@@ -213,7 +212,6 @@ func logVllmCompletionRequest(log *zap.Logger, cr *vllm.CompletionRequest, prod,
 
 	if prod {
 		fields := []zapcore.Field{
-			zap.String(logFiledNameCorrelationId, cid),
```
Comment (spikelu2016) [inline internal/server/web/proxy/vllm.go:216]: cid can br removed from the function header here and the `logVllmChatCompletionRequest` function

File: internal/server/web/proxy/thread.go [hunk 3]
```diff
@@ -59,7 +57,6 @@ func logThreadResponse(log *zap.Logger, data []byte, prod bool, cid string) {
 func logRetrieveThreadRequest(log *zap.Logger, prod bool, cid, tid string) {
```
Comment (spikelu2016) [inline internal/server/web/proxy/thread.go:57]: cid can be removed from the function headers for all the functions in this file

File: internal/server/web/proxy/models.go [hunk 4]
```diff
@@ -61,7 +58,6 @@ func logRetrieveModelResponse(log *zap.Logger, data []byte, prod bool, cid strin
 func logDeleteModelRequest(log *zap.Logger, prod bool, cid, model string) {
```
Comment (spikelu2016) [inline internal/server/web/proxy/models.go:58]: cid can be removed from the function headers for all the functions in this file

**Comment by spikelu2016** [comment]
unused `cid` can still be found across many functions. you can find all of them using vscode search. removing all of them would be great

**Comment by GingerMoon** [comment]
all cids are removed.

**Review by spikelu2016 — APPROVED** [review]

**Pull request merged**